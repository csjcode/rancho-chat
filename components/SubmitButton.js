import React from 'react'
import { StyleSheet, TouchableOpacity, Text } from 'react-native'
import colors from '../constants/colors'
import getColors from '../constants/getColors'
const colorsTheme = getColors()

const SubmitButton = (props) => {
  const enabledBgColor = props.color || colorsTheme.primary
  const disabledBgColor = colorsTheme.lightGrey
  const bgColor = props.disabled ? disabledBgColor : enabledBgColor

  return (
    <TouchableOpacity
      onPress={props.disabled ? () => {} : props.onPress}
      style={{
        ...styles.button,
        ...props.style,
        ...{ backgroundColor: bgColor },
      }}
    >
      <Text
        style={{
          color: props.disabled
            ? colorsTheme.grey
            : colorsTheme.submitButtonTextColor,
        }}
      >
        {props.title}
      </Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    paddingHorizontal: 30,
    paddingVertical: 10,
    borderRadius: 30,
    justifyContent: 'center',
    alignItems: 'center',
    // color: 'black',
  },
})

export default SubmitButton
