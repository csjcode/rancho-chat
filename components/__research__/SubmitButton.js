/**
 * 
 * 
 * 
//Unit Test Code
import 'react-native';
import React from 'react';
import SubmitButton from './SubmitButton';

import renderer from 'react-test-renderer';

describe('SubmitButton component', () => {

  it('renders correctly', () => {
    const tree = renderer
    .create(<SubmitButton title='Submit' />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with given props', () => {
    const tree = renderer
    .create(<SubmitButton title='Submit' disabled={true} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with given color', () => {
    const tree = renderer
    .create(<SubmitButton title='Submit' color='red' />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with given style', () => {
    const tree = renderer
    .create(<SubmitButton title='Submit' style={{margin: 10}} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls the onPress function', () => {
    const onPressMock = jest.fn();
    const tree = renderer
    .create(<SubmitButton title='Submit' onPress={onPressMock} />)
    .toJSON();
    expect(onPressMock).toHaveBeenCalledTimes(0);
    tree.props.onPress();
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

});

//These tests check that the SubmitButton component renders correctly with the given props, such as title, disabled, color and style. It also checks that when the onPress function is called, it is indeed called. In addition, the tests also check that the component renders correctly with snapshots.


 */
