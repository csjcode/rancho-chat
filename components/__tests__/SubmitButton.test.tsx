import React from "react";
import renderer from "react-test-renderer";
import { TouchableOpacity, Text } from "react-native";
import SubmitButton from "../SubmitButton";

describe("Testing SubmitButton component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<SubmitButton title="Submit" onPress={() => {}} style={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with disabled flag", () => {
    const tree = renderer
      .create(
        <SubmitButton title="Submit" disabled onPress={() => {}} style={{}} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
