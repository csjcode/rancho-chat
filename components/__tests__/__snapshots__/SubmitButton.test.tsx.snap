// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SubmitButton component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00ffbd",
      "borderRadius": 30,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 30,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1b4e3f",
      }
    }
  >
    Submit
  </Text>
</View>
`;

exports[`Testing SubmitButton component renders correctly with disabled flag 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#bdc3c7",
      "borderRadius": 30,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 30,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#7f8c8d",
      }
    }
  >
    Submit
  </Text>
</View>
`;
